dockstore-webservice/src/main/java/io/dockstore/webservice/resources/proposedGA4GH/ToolsApiExtendedServiceImpl.java

```diff
@@ -391,6 +409,57 @@ public Response setSourceFileMetadata(String type, String id, String versionId,
         throw new CustomWebApplicationException("Could not submit verification information", HttpStatus.SC_BAD_REQUEST);
     }
 
+    @Override
+    public Response submitMetricsData(String id, String versionId, Partner platform, User owner, String description, List<Execution> executions) {
+        // Check that the entry and version exists
+        Entry<?, ?> entry;
+        try {
+            entry = getEntry(id, Optional.of(owner));
+        } catch (UnsupportedEncodingException | IllegalArgumentException e) {
+            return BAD_DECODE_REGISTRY_RESPONSE;
+        }
+
+        Optional<? extends Version<?>> version = getVersion(entry, versionId);
+        if (entry == null) {
+            return Response.status(Response.Status.NOT_FOUND.getStatusCode(), "Tool not found").build();
+        } else if (version.isEmpty()) {
+            return Response.status(Response.Status.NOT_FOUND.getStatusCode(), "Version not found").build();
+        }
+
+        // Check that all executions have at least the ExecutionStatus
+        if (executions.stream().anyMatch(execution -> execution.getExecutionStatus() == null)) {
```

Comment:
What would it look like if someone wanted to submit only `additionalProperties` 
For example, DNAstack only submitting validation information, noting that some workflows may be non-executable with semantic issues. I suppose they could rig up a fail for something that is non-valid, but what about if they only want to report passing validation? 
Another example might be some already batched information, for example a platform didn't want to report cost per execution but was able to tell us that 1000 runs on average costed $5

Reply:
> For example, DNAstack only submitting validation information, noting that some workflows may be non-executable with semantic issues. I suppose they could rig up a fail for something that is non-valid, but what about if they only want to report passing validation?

I had created `FAILED_SEMANTIC_INVALID` here as an `ExecutionStatus` [here](https://github.com/dockstore/dockstore/blob/f7c880570edb5c8b6c11f61dedf520e191fbcd81/dockstore-webservice/src/main/java/io/dockstore/webservice/core/metrics/ExecutionStatusCountMetric.java#L128-L129) and I imagined that they would provide that as the `ExecutionStatus` for a workflow with semantic issues

> I suppose they could rig up a fail for something that is non-valid, but what about if they only want to report passing validation?

Not sure if I'm following. Are you saying that they don't want to report execution status and they would only want to report validation? I had assumed that they go hand-in-hand: succcessful execution = valid workflow and failed = invalid

Reply:
Actually, I forgot about/missed the invalid status, so the second question doesn't make sense anyway